sway-lib-core/src/never.sw

```diff
@@ -0,0 +1,11 @@
+library never;
+
+use ::ops::Not;
+
+pub enum Never {}
+
+impl Not for Never {
+    fn not(self) -> Self {
```

Comment:
Are there other operations we need to implement for `Never`?

Reply:
Rust has the following [trait implementations](https://doc.rust-lang.org/std/primitive.never.html#trait-implementations) for ! (Never):
`Clone`
`Debug`
`Display`
`Error`
`Hash`
`Not`
`Ord`
`PartialEq`
`PartialOrd`
`Termination`
`Copy`
`Eq`
From the list above we already have in std lib: `Not` `Ord` `Eq`
We indeed get an error `No method named "eq" found for type "Never".` so I will add the traits `Eq` and `Òrd` and corresponding tests.